pkg/endpoint/endpoint.go

```diff
@@ -985,6 +1121,10 @@ func ParseEndpoint(strEp string) (*Endpoint, error) {
 
 	ep.state = StateRestoring
 
+	ep.Mutex.Lock()
+	defer ep.Mutex.Unlock()
+	ep.runK8sCiliumEndpointSync()
```

Comment:
What's the reason to execute `runK8sCiliumEndpointSync` here?

Reply:
Isn't this how restored endpoints are built? we don't have a single NewEndpoint function so we need to start the controller in all places we create an endpoint. Should I do it somewhere else for those?